tests/rust.rs

```diff
@@ -598,6 +598,16 @@ fn tuple_list_as_map_vecdeque() {
     );
 }
 
+#[test]
+fn test_string_empty_as_none() {
+    #[derive(Debug, PartialEq, Deserialize, Serialize)]
+    struct S(#[serde(with = "serde_with::rust::string_empty_as_none", default)] Option<String>);
```

Comment:
```suggestion
    struct S(#[serde(with = "serde_with::rust::string_empty_as_none")] Option<String>);
```

The `default` should not be there, otherwise you tell serde that this field can also be missing.